File: test_hash.py
import json
import httpx
import time

def fetch_and_clean():
    resp = httpx.get("http://127.0.0.1:8014/api/plazas_status")
    data = resp.json()["plazas"]
    for p in data:
        for a in p.get("agents", []):
            a.pop("last_active", None)
            a.pop("login_history", None)
    return data

if __name__ == "__main__":
    d1 = fetch_and_clean()
    time.sleep(3)
    d2 = fetch_and_clean()
    if json.dumps(d1) != json.dumps(d2):
        print("Differences found!")
        print(d1 == d2) # Deep equality check
    else:
        print("No differences.")

File: test_hash.js
const fs = require('fs');
function stripVolatileAgentFields(agent) {
    if (!agent || typeof agent !== 'object') return agent;
    const cloned = { ...agent, card: { ...(agent.card || {}) } };
    delete cloned.last_active;
    delete cloned.login_history;
    return cloned;
}
async function getHash() {
    const res = await fetch("http://127.0.0.1:8014/api/plazas_status");
    const data = await res.json();
    const plaza = data.plazas[0];
    const isOnline = !!plaza.online;
    const card = plaza.card || {};
    const showInactive = true;
    const sortOrder = "active-new";
    const agents = plaza.agents || [];
    // sortAgents
    agents.sort((a,b) => b.last_active - a.last_active);
    const nextHash = JSON.stringify({
        key: plaza.url,
        online: !!isOnline,
        card,
        showInactive: !!showInactive,
        sortOrder,
        agents: agents.map(stripVolatileAgentFields)
    });
    return nextHash;
}
async function run() {
    const h1 = await getHash();
    console.log("Waiting 3s...");
    await new Promise(r => setTimeout(r, 3000));
    const h2 = await getHash();
    console.log("h1 === h2?", h1 === h2);
    if (h1 !== h2) {
        fs.writeFileSync("h1.json", h1);
        fs.writeFileSync("h2.json", h2);
    }
}
run();
